src/systems/settling.cairo

```diff
@@ -0,0 +1,289 @@
+                    rivers: realm_data.rivers,
+                    regions: realm_data.regions,
+                    wonder: realm_data.wonder,
+                    order: realm_data.order,
+                    }, Owner {
+                    address: owner.address, 
+                    }, Age {
+                    born_at: starknet::get_block_timestamp(), 
+                }
+            )
+        );
+        // mint base resources for the realm
+        let resource_ids: Span<u8> = unpack_resource_ids(
+            realm_data.resource_ids_packed, realm_data.resource_ids_count
+        );
+        let mut index = 0_usize;
+        loop {
+            if index == realm_data.resource_ids_count.into() {
+                break ();
+            };
+            let resource_id: u8 = *resource_ids[index];
+            let resource_id_felt: felt252 = resource_id.into();
+            let resource_query: Query = (realm_id, resource_id_felt).into();
+            commands::<Resource>::set_entity(
+                resource_query,
+                (Resource { id: resource_id, balance: config.base_resources_per_day,  })
+            );
+            index += 1;
+        }
+
+        // transfer Realm ERC721 to world contract
+        erc721.transfer_from(owner.address, world_address, realm_id, );
+    }
+}
+
+#[system]
+mod Unsettle {
+    use traits::Into;
+
+    use eternum::constants::WORLD_CONFIG_ID;
+    use eternum::interfaces::{IERC721Dispatcher, IERC721DispatcherTrait};
+    use eternum::erc721::erc721::Position;
+    use eternum::components::owner::Owner;
+    use eternum::components::realm::Realm;
+    use eternum::components::resources::Resource;
+    use eternum::components::age::Age;
+    use eternum::components::config::WorldConfig;
+
+    fn execute(realm_id: felt252) {
+        // get the ERC721 contract
+        let config = commands::<WorldConfig>::entity(WORLD_CONFIG_ID.into());
+        let token = config.realm_l2_contract;
+
+        // get the owner
+        let realm_query: Query = realm_id.into();
+        let owner = commands::<Owner>::entity(realm_query);
+        // TODO: wait for set_account_contract_address to be available
+        // let caller = starknet::get_tx_info().unbox().account_contract_address;
+        // assert caller is owner
+        // assert(owner.address == caller, 'Only owner can unsettle');
+
+        // transfer back nft from world to owner
+        IERC721Dispatcher { contract_address: token }.transfer_from(
+            world_address,
+            owner.address,
+            realm_id,
+        );
+
+        // delete entity
+        let world = IWorldDispatcher {contract_address: world_address};
+        world.delete_entity('Owner'.into(), realm_query);
+        world.delete_entity('Position'.into(), realm_query);
+        world.delete_entity('Realm'.into(), realm_query);
+        world.delete_entity('Age'.into(), realm_query);
+        // TODO: should delete resources ?
```

Comment:
I guess it's not necessary because the resource entities will get re-set on settling again.